coredb-operator/src/controller.rs

```diff
@@ -146,8 +148,10 @@ impl CoreDB {
             debug!("Did not find primary pod");
             return Ok(Action::requeue(Duration::from_secs(1)));
         }
-        if !is_pod_ready().matches_object(Some(&primary_pod.unwrap())) {
-            debug!("Did not find primary pod is ready");
+        let primary_pod = primary_pod.unwrap();
+
+        if !is_postgres_ready().matches_object(Some(&primary_pod)) {
```

Comment:
we wait just for postgres to be ready, not the whole pod, before creating the postgres exporter role